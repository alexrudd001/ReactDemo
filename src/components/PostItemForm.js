import React, { useState } from 'react';

const PostItemForm = (props) => {
    const [ title, setTitle ] = useState(props.title ? props.title : '');
    const [ body, setBody ] = useState(props.body ? props.body : '');
    const [ error, setError ] = useState('');
    const submitHandler = (e) => {
        e.preventDefault();
        if (!title || !body) {
            setError('Please provide title and body.');
        } else {
            setError('');
            props.onSubmit({
                title,
                body
            });
        }
    };
    return (
        <div>
            <h3>Enter Details</h3>
            <form onSubmit={submitHandler}>
                <label htmlFor="title">Title: </label>
                <br />
                <input type="text" name="title" id="title" className="textBox" onChange={(e) => setTitle(e.target.value)} value={title} />
                <br />
                <br />
                <label htmlFor="body">Body: </label>
                <br />
                <textarea name="body" id="body" className="textAreaBox" onChange={(e) => setBody(e.target.value)} value={body} />
                <br />
                <span className="error">{error}</span>
                <br />
                <br />
                <input type="submit" value="Submit" className="submit" />
            </form>
        </div>
    );
};

export default PostItemForm;
