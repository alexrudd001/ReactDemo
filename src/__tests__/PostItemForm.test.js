import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostItemForm from '../components/PostItemForm';

test('renders PostItemForm correctly', () => {
    const wrapper = shallow(<PostItemForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test('renders PostItemForm correctly with post data', () => {
    const wrapper = shallow(<PostItemForm title={'post title'} body={'post body'} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test('call onSubmit prop for valid form submission', () => {
    const onSubmitSpy = jest.fn();
    const wrapper = shallow(<PostItemForm title={'post title'} body={'post body'} onSubmit={onSubmitSpy} />);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        title: 'post title',
        body: 'post body'
    });
});

test('renders error for invalid form submission', () => {
    const wrapper = mount(<PostItemForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(wrapper.find('span').text()).toEqual('Please provide title and body.');
});
