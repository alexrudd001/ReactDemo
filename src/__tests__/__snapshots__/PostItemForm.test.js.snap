// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PostItemForm correctly 1`] = `
<div>
  <h3>
    Enter Details
  </h3>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="title"
    >
      Title: 
    </label>
    <br />
    <input
      className="textBox"
      id="title"
      name="title"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <br />
    <label
      htmlFor="body"
    >
      Body: 
    </label>
    <br />
    <textarea
      className="textAreaBox"
      id="body"
      name="body"
      onChange={[Function]}
      value=""
    />
    <br />
    <span
      className="error"
    />
    <br />
    <br />
    <input
      className="submit"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;

exports[`renders PostItemForm correctly with post data 1`] = `
<div>
  <h3>
    Enter Details
  </h3>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="title"
    >
      Title: 
    </label>
    <br />
    <input
      className="textBox"
      id="title"
      name="title"
      onChange={[Function]}
      type="text"
      value="post title"
    />
    <br />
    <br />
    <label
      htmlFor="body"
    >
      Body: 
    </label>
    <br />
    <textarea
      className="textAreaBox"
      id="body"
      name="body"
      onChange={[Function]}
      value="post body"
    />
    <br />
    <span
      className="error"
    />
    <br />
    <br />
    <input
      className="submit"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;

exports[`renders error for invalid form submission 1`] = `
<PostItemForm>
  <div>
    <h3>
      Enter Details
    </h3>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="title"
      >
        Title: 
      </label>
      <br />
      <input
        className="textBox"
        id="title"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
      <br />
      <br />
      <label
        htmlFor="body"
      >
        Body: 
      </label>
      <br />
      <textarea
        className="textAreaBox"
        id="body"
        name="body"
        onChange={[Function]}
        value=""
      />
      <br />
      <span
        className="error"
      />
      <br />
      <br />
      <input
        className="submit"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</PostItemForm>
`;
